lib/glific/providers/gupshup/message.ex

```diff
@@ -191,27 +192,22 @@ defmodule Glific.Providers.Gupshup.Message do
     references = get_in(params, ["payload", "references"])
     deductions = get_in(params, ["payload", "deductions"])
     bsp_message_id = references["gsId"] || references["id"]
-    phone = references["destination"]
-
-    Repo.fetch_by(Message, %{
-      bsp_message_id: bsp_message_id
-    })
-    |> case do
-      {:ok, message} ->
-        message_conversation = %{
-          deduction_type: deductions["type"],
-          is_billable: deductions["billable"],
-          conversation_id: references["conversationId"],
-          payload: params,
-          message_id: message.id
-        }
-
-        {:ok, message_conversation}
-
-      _ ->
-        error = "Could not find message with id: #{bsp_message_id} and phone #{phone}"
-        Glific.log_error(error, false)
-    end
+
+    message =
+      Message
+      |> where([m], m.bsp_message_id == ^bsp_message_id)
+      |> select([m], %{id: m.id})
+      |> Repo.all(skip_organization_id: true)
```

Comment:
Why do we need skip_organization_id: true here?

Reply:
Let's use
Repo.fetch_by(Message, %{
      bsp_message_id: bsp_message_id
    }) with case statement only

Reply:
@pankaj-ag I have updated the query. Thanks